database/dialect.go

```diff
@@ -135,3 +135,13 @@ func (s *store) ListMigrations(
 	}
 	return migrations, nil
 }
+
+func (s *store) GetLatestVersion(ctx context.Context, db DBTxConn) (int64, error) {
```

Comment:
`max` functions usually return `NULL` when nothing is found, which is something we should guard against. I think you'll want something like this.

```go
func (s *store) GetLatestVersion(ctx context.Context, db DBTxConn) (int64, error) {
	q := s.querier.GetLatestVersion(s.tablename)
	var maxVersion sql.NullInt64
	err := db.QueryRowContext(ctx, q).Scan(&maxVersion)
	if err != nil {
		return 0, fmt.Errorf("failed to get last version: %w", err)
	}
	if maxVersion.Valid {
		return maxVersion.Int64, nil
	}
	return -1, ErrVersionNotFound
}
```

A test like this should help cover most of the edge cases across postgres / sqlite, and we'll need to make sure it works like that across the rest of the databases.

```go
t.Run("GetLatestVersion", func(t *testing.T) {
	dir := t.TempDir()
	db, err := sql.Open("sqlite", filepath.Join(dir, "sql_embed.db"))
	check.NoError(t, err)
	store, err := database.NewStore(database.DialectSQLite3, "foo")
	check.NoError(t, err)
	err = store.CreateVersionTable(context.Background(), db)
	check.NoError(t, err)
	v, err := store.GetLatestVersion(context.Background(), db)
	check.NoError(t, err)
	_ = v // ??
})
```

Reply:
Added tests for different databases
[cb36bc6](https://github.com/pressly/goose/pull/675/commits/cb36bc6eb1b6127f40461221a000dad6c7b3cee1)